### PR TITLE
add leaky_relu torch frontend 

### PR DESCRIPTION
**Pull request by AymenSe**
I still have some issues with the testing because the function name (leaky_relu) does not exist as a key:
`frontend_ret = frontend_fw.__dict__[fn_name](*args_frontend, **kwargs_frontend)`

As you can see below:

image

Could anyone help me to tackle this issue?
I really appreciate any help you can provide.

### CODE REVIEW

**Comment by AymenSe** [comment]
- [ ] leaky_relu_ #2724
- [ ] relu_  #2667
- [ ] leaky_relu #2723

**Review by Daniel4078 — CHANGES_REQUESTED** [review]
I suppose that these two arguments "as_variable_flags" and "native_array_flags" of helpers.test_function should not be generated randomly. For example, "native_array_flags" tells test_function if the other arguments feeding into the leaky_relu function should be treated as native arrays during the tests. Please update the test. P.S. I did not see any updates about "relu_"  and "leaky_relu_" functions, then why you relate this pull request with those issues?

**Comment by AymenSe** [comment]
@Daniel4078 I have fixed the flags by setting them to false, but still the same problem! the test cannot find ` fn_name = 'leaky_relu'` in `frontend_fw.__dict__[fn_name]` and once I printed the `frontend_fw.__dict__` in order to check what inside this, I found most of the functions keys (relu, rrelu, sigmoid ..etc) but there isn't a key with the name 'leaky_relu'.

**Comment by Daniel4078** [comment]
currently, the key you are looking for is call "torch.nn.functional.leaky_relu", see here for more details: https://lets-unify.ai/ivy/deep_dive/17_ivy_frontends_tests.html.

**Review by Daniel4078 — CHANGES_REQUESTED** [review]
Please try remove your change to the .idea files, there is a guide about how to do that: https://devconnected.com/how-to-remove-files-from-git-commit
Also, please refer to this example of hypothesis test to see what to do about the native_array flag and with_out flag, though it is not a front-end test:
# execute_with_gradients
@given(
    fun=lambda xs: ivy.nested_indices_where(nest=xs, fn=lambda _: True, ),
    dtypes_and_xs=helpers.dtype_and_values(),
    retain_grads=st.booleans(),
    container_flag=st.booleans(),
    num_positional_args=helpers.num_positional_args(fn_name="execute_with_gradients"),
    data=st.data()
)
def test_execute_with_gradients(
        *,
        fun,
        dtype_and_xs,
        retain_grads,
        container_flag,
        num_positional_args,
        native_array,
        instance_method,
        fw,
):
    dtypes, xs = dtype_and_xs
    helpers.test_function(
        input_dtypes=dtypes,
        as_variable_flags=[False, True, False],
        with_out=False,
        num_positional_args=num_positional_args,
        native_array_flags=native_array,
        container_flags=container_flag,
        instance_method=instance_method,
        fw=fw,
        fn_name="execute_with_gradients",
        func=fun,
        xs=xs,
        retain_grads=retain_grads
    )
Thank you.

**Comment by AymenSe** [comment]
Sorry, I accidentally closed this PR!

**Comment by AymenSe** [comment]
This is my new PR #2914 you can check it and I suppose that everything is working fine. 
 @Daniel4078 @1Doomdie1 @sreegandh804
